server/channels/app/channel.go

```diff
@@ -222,6 +228,15 @@ func (a *App) RenameChannel(c request.CTX, channel *model.Channel, newChannelNam
 }
 
 func (a *App) CreateChannel(c request.CTX, channel *model.Channel, addMember bool) (*model.Channel, *model.AppError) {
+	if *a.Config().ExperimentalSettings.ExperimentalChannelCategorySorting {
+		// Handle category names in channel name
+		if strings.Contains(channel.DisplayName, "/") {
```

Comment:
Below you combine both the if conditions with &&, but here they are separate. Let's just use one approach.

Also, would recommend to refactor out the displayName and CategoryName splitting logic to somewhere else. This feels like it should be coming from a separate, properly named function.